### DISCUSSION TITLE
Can I reverse-engineer an image slider?

### DISCUSSION THREAD

**Discussion by sophielamb** [thegetty/quire#152] [Get Help]
Hi Quire folks,

I would like to create and embed an image slider in our publication, which ideally would be implemented in the same way as a normal figure embed. There are not many examples out there to work from, and I am not familiar with using Hugo. 

I can make sliders in HTML and CSS, and of course there are a shed-load of slider snippets out there. What I can't get my head around is marrying up an HTML/CSS slider with creating a shortcode file. How doable is this?

If anyone has pointers I'd be very grateful. 

Sophie

**Comment by Erin-Cecele** [comment]
Hi @sophielamb. Just checking in. Have you made any progress on this?

**Reply by sophielamb** [reply]
Hi @Erin-Cecele, thanks so much for keeping tabs on me!

I've made fair progress with building this shortcode. I found a handful of tutorials about Hugo, and pulled apart the Quire template shortcodes to see how they work. I used one as a starter to drop in the code, so I have a reusable image comparison slider. What I need to do next is learn how to add parameters to add figure text to the `<span>` and to pull in target images into `<img>`. For now I have placeholder text and images:

    <figure class="cd-image-container">
      <img src="https://images.unsplash.com/photo-1499162526249-f5b8b9ba9923?auto=format&fit=crop&w=1050&q=80" alt="original image"/>
      <span class="cd-image-label" data-type="original">Original</span>
      <div class="cd-resize-img"> 
        <img src="https://images.unsplash.com/photo-1502290326793-2039d3b222f4?auto=format&fit=crop&w=1050&q=80" alt="modified image"/>
        <span class="cd-image-label" data-type="modified">Modified</span>
      </div>
      <span class="cd-handle"></span>
    </figure>

Also, one problem I'm having is that the span containing the 'Original' label on the lower image disappears once the slider runs over it.

This wasn't happening when I had the code prior to putting it into Hugo/Quire. I think it relates to a 'hidden' class which is activated by JQuery but it's not being deactivated once the slide button is dragged left again. I have to admit this is beyond my skill-set and the code I'm using was copied from a tutorial:

      function updateLabel(label, resizeElement, position) {
        if (position == "left") {
          label.offset().left + label.outerWidth() <
          resizeElement.offset().left + resizeElement.outerWidth()
            ? label.removeClass("is-hidden")
            : label.addClass("is-hidden");
        } else {
          label.offset().left >
          resizeElement.offset().left + resizeElement.outerWidth()
            ? label.removeClass("is-hidden")
            : label.addClass("is-hidden");
        }
      }

Any help is hugely appreciated - thanks!
